### PR TITLE
setup travis

### PR DESCRIPTION
**Pull request by glensc**
setup travis for linux-pam project, contributors' PR's could be tested automatically.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
+sudo: required
+dist: precise
```
Comment (ldv-alt) [inline .travis.yml]: Why precise?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            - ubuntu-toolchain-r-test
+          packages:
+            - gcc-4.9
+      env: COMPILER=gcc-4.9
+    - compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
```
Comment (ldv-alt) [inline .travis.yml]: The oldest supported gcc version is 5.4, why bother testing older versions?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            - gcc-5
+      env: COMPILER=gcc-5
+    - compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - gcc-6
+      env: COMPILER=gcc-6
```
Comment (ldv-alt) [inline .travis.yml]: gcc-7 is also worth testing

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      env: COMPILER=clang-3.6
+    - compiler: clang
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.7
+          packages:
+            - clang-3.7
+      env: COMPILER=clang-3.7
```
Comment (ldv-alt) [inline .travis.yml]: i wouldn't bother testing clang versions older than 3.8.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+install:
+  - sudo apt-get -qq update
+  - sudo apt-get -qq install automake autopoint gettext xsltproc docbook-xml docbook-xsl
```
Comment (ldv-alt) [inline .travis.yml]: why not addons.apt.packages?
why listing already installed packages?

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+script:
+  - $COMPILER --version
+  - autopoint --version
+  - git fetch --tags
+  - git revert -n 58d2da7160a477a5f5654be08bb36cc475eb75d7
```
Comment (ldv-alt) [inline .travis.yml]: this is no-no, gettext 0.18.3 is the latest oldstable
if the build environment does not provide a decent version of gettext, fix the environment

File: .travis.yml (hunk 1 above)
Comment (glensc) [inline .travis.yml]: there's no requirements, so i just took something that works (based on logrotate project setup)

File: .travis.yml (hunk 2 above)
Comment (glensc) [inline .travis.yml]: don't know what is the oldest supported gcc, just took something as base to get things going.

File: .travis.yml (hunk 5 above)
Comment (glensc) [inline .travis.yml]: can't mix matrix in sane way, i.e having per item packages and extending with general requirements. have to duplicate them.

but then again, this is example which can be improved if really needed

File: .travis.yml (hunk 6 above)
Comment (glensc) [inline .travis.yml]: as i wrote in PR body, i do not know how to fix this. so it's temporary measure to get travis working

> don't know how to pull newer gettext in travis
> so it just reverts 58d2da7 during the build.

**Comment by glensc** [comment]
i can update compiler dependencies, but the rest of the changes can be done post merge and by other people.

i just wanted to help, but now it feels like i'm being attacked that i did things so wrong even there aren't any requirements specified anywhere.

File: .travis.yml (hunk 1 above)
Comment (ldv-alt) [inline .travis.yml]: precise is ubuntu 12.04, trusty is 14.04; the reason why you needed that gettext hack is that you've chosen the wrong repository

File: .travis.yml (hunk 2 above)
Comment (ldv-alt) [inline .travis.yml]: it's a good idea to visit https://gcc.gnu.org/ then :)

**Comment by ldv-alt** [comment]
> i can update compiler dependencies, but the rest of the changes can be done post merge and by other people.
Thanks for the idea of setting up travis for linux-pam.
Unfortunately, your proposal lacks the quality required by linux-pam.
We usually don't apply changes that are known to need fixes.

File: .travis.yml (hunk 5 above)
Comment (glensc) [inline .travis.yml]: also, could you more precise? which packages are already installed packages?

**Comment by glensc** [comment]
@ldv-alt changes made

**Comment by steven-joruk** [comment]
I've made the remaining changes recommended by @ldv-alt. Please can you review and cherry-pick [my commit](https://github.com/steven-joruk/linux-pam/commit/db281b583823e6170c5780341cbf597c03ff8ce4) in to your pull request, @glensc?

Here's the Travis CI build from my branch - https://travis-ci.org/steven-joruk/linux-pam

**Comment by glensc** [comment]
@steven-joruk sure, thanks!

**Pull request closed without merging**